src/pineko/parser.py

```diff
@@ -35,7 +71,54 @@ def _load_yaml(yaml_file):
     return ret
 
 
-def get_yaml_information(yaml_file, grids_folder):
+def _apfelcomb_compatibility_flags(gridpaths, metadata):
+    """
+    Prepare the apfelcomb-pineappl compatibility fixes
+
+    Returns
+    -------
+        apfelcomb_norm: np.array
+            Per-point normalization factor to be applied to the grid
+            to be compatible with the data
+        apfelcomb_repetition_flag: bool
+            Whether the fktable is a single point which gets repeated up to a certain size
+            (for instance to normalize a distribution)
+        shift: list(int)
+            Shift in the data index for each grid that forms the fktable
+    """
+    operands = metadata["operands"]
+    apfelcomb_norm = None
+    if norms := metadata.get("apfelcomb_norm"):
+        # There's no easy way for an fktable to know its role in given operation:
+        for factors, grids in zip(norms, operands):
```

Comment:
Please,  `raise` if `len(norms) != len(operands)`,  `zip()` will silently cut one of the two, and this is an expected behavior for the user of the function.

Reply:
No, if `len(norms) != len(operators)` that's fine. It will continue its path. I can make it not enter the loop if that's the case if you prefer it.

Reply:
If `len(norms) == len(operators) - 1` because of a typo, you will have one element less used in `operators` silently...

Reply:
If there's a typo in the yaml file I cannot do anything. I can add checks for warnings for weird stuff, but the problem is the apfelcomb stuff are afterthoughts so they _can_ be weird stuff.
I can try to look through the yaml db to improve on them though. It will also fix some of the other issues you raised...